packages/expo-modules-autolinking/android/expo-gradle-plugin/expo-autolinking-plugin/src/main/kotlin/expo/modules/plugin/ExpoRootProjectPlugin.kt

```diff
@@ -0,0 +1,44 @@
+package expo.modules.plugin
+
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+import org.gradle.api.artifacts.VersionCatalog
+import org.gradle.api.artifacts.VersionCatalogsExtension
+import org.gradle.api.plugins.ExtraPropertiesExtension
+import org.gradle.internal.extensions.core.extra
+import java.util.Optional
+import kotlin.jvm.optionals.getOrNull
+
+class ExpoRootProjectPlugin : Plugin<Project> {
+  override fun apply(rootProject: Project) {
+    val versionCatalogs = rootProject.extensions.getByType(VersionCatalogsExtension::class.java)
+    val libs = versionCatalogs.find("libs")
+
+    with(rootProject) {
+      defineDefaultProperties(libs)
+    }
+  }
+}
+
+fun Project.defineDefaultProperties(versionCatalogs: Optional<VersionCatalog>) {
+  // Android related
+  extra.setIfNotExist("buildToolsVersion") { versionCatalogs.getVersionOrDefault("buildTools", "35.0.0") }
+  extra.setIfNotExist("minSdkVersion") { Integer.parseInt(versionCatalogs.getVersionOrDefault("minSdk", "24")) }
+  extra.setIfNotExist("compileSdkVersion") { Integer.parseInt(versionCatalogs.getVersionOrDefault("compileSdk", "35")) }
+  extra.setIfNotExist("targetSdkVersion") { Integer.parseInt(versionCatalogs.getVersionOrDefault("targetSdk", "35")) }
+  extra.setIfNotExist("ndkVersion") { versionCatalogs.getVersionOrDefault("ndkVersion", "27.1.12297006") }
+
+  // Kotlin related
+  extra.setIfNotExist("kotlinVersion") { versionCatalogs.getVersionOrDefault("kotlin", "2.0.21") }
+  extra.setIfNotExist("kspVersion") { versionCatalogs.getVersionOrDefault("ksp", "2.0.21-1.0.28") }
```

Comment:
We can use the ksp lookup map here instead of hardcoding it. 
In the long term, I would remove default values here and throw an error if the version catalogs aren't present.